kubetest/kops.go

```diff
@@ -487,3 +498,47 @@ func (k *kops) runKopsDump() (*kopsDump, error) {
 
 	return dump, nil
 }
+
+// kops deployer implements publisher
+var _ publisher = &kops{}
+
+// Publish will publish a success file, it is called if the tests were successful
+func (k kops) Publish() error {
+	if k.kopsPublishVersion == "" {
+		return nil
+	}
+
+	if k.kopsVersion == "" {
```

Comment:
It cannot figure this out from the state of the kops source?

Reply:
We could (probably!) find the version, but we can't easily find the absolute URL path.  We could write a marker file when we download it, but I'm not sure if that's any cleaner.

This only works currently when we download a kops binary, which is the common case.  In theory we could have got a kops binary we built ourselves (e.g. on a PR builder).  AFAIK, the only one of those jobs that writes a "success file" is `ci-kops-build` (the bottom turtle), which calls a target in the kops makefile which writes that file directly.